### ISSUE TITLE
The distribution of electromagnetic field caused by semi-solution is different from that of full-solution.

### ISSUE THREAD

**Issue by xungu1** [kc-ml2/meent#86]
Hello, based on the help you provided before, I deduced the semi-solution of RCWA and the formula of bilateral light source illumination, but there was a little problem in the semi-solution. When the semi-solution method is used to calculate Rs, Rp or Ts, Tp, the results of the semi-solution method and the full solution method are exactly the same, but the propagation coefficients c1plus, c1minus, c2plus and c2minus have some problems. They can be numerically the same as the calculation results of the full solution, but the corresponding variables are different, which also leads to the final calculated electric field distribution and magnetic field distribution different from the full solution ( this result will be shown in the pdf provided by me ). Is it caused by ignoring what in the process of my formula derivation ?
[Partial solution.pdf](https://github.com/user-attachments/files/18257050/Partial.solution.pdf)

**Comment by yonghakim** [comment]
```python
    # Final Equation in form of AX=B
    final_A = np.block(
        [
            [I, O, -big_F_11, -big_F_12],
            [O, -1j * Kz_top / (n_top ** 2), -big_F_21, -big_F_22],
            [-1j * Kz_top, O, -big_G_11, -big_G_12],
            [O, I, -big_G_21, -big_G_22],
        ]
    )

    final_B = np.block(
        [
            [-np.sin(psi) * delta_i0],
            [np.cos(psi) * np.cos(theta) * delta_i0],
            [-1j * np.sin(psi) * n_top * np.cos(theta) * delta_i0],
            [-1j * n_top * np.cos(psi) * delta_i0]
        ]
    )
```

The formulation in meent is different from the paper.
Please use above and retry. I think this will match.

**Comment by xungu1** [comment]
I have tried the new formula given by you. Although the calculation results of Rs, Rp, Ts and Tp of the new formula are different from those of the old formula, the calculation results of the partial-solution method of the new formula are exactly the same as those of the full-solution method, but there are still problems in the propagation coefficient. In this application, the new formula has different symbols except for the corresponding variables of the propagation coefficient of the partial-solution. The final calculated electromagnetic field distribution is still different. ( This result will be shown in the pdf provided by me )
[Uploading new formula relust.pdf…]()

**Comment by yonghakim** [comment]
The notation you are using is different from meent. Please see equ 40 to 44 in meent paper. See the order and definition of coefficients.

**Comment by xungu1** [comment]
The formula I used in the semi-solution is rewritten according to ( 49-50 ) in your paper. In order to facilitate the solution, I changed the original equation 2 into equation 4, equation 3 into equation 2, equation 4 into equation 3, only the order of the equations has changed. This operation should not affect the results.
![image](https://github.com/user-attachments/assets/fc8a59a8-36ad-4339-8d36-7e7c2d5838a8)
